controllers/projectController.go

```diff
@@ -0,0 +1,108 @@
+package controllers
+
+import (
+	. "clokify/models"
+	. "clokify/services"
+	"encoding/json"
+	"io"
+	"log"
+	"strconv"
+
+	. "clokify/types"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+func CreateProject(srvMananger *ServiceManager, db *gorm.DB) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var project ProjectCreateType
+		projectService := &ProjectServiceManager{
+			ServiceManager: srvMananger,
+		}
+
+		bodyData, err := io.ReadAll(ctx.Request.Body)
+		if err != nil {
+			ctx.JSON(400, gin.H{
+				"error": "Failed to read request body",
+			})
+			return
+		}
+
+		if err := json.Unmarshal(bodyData, &project); err != nil {
+			ctx.JSON(400, gin.H{
+				"error": "Invalid request body",
+			})
+			return
+		}
+
```

Comment:
we are sending the whole object and another parameter is the id of that object. can you cater to the project object or id parameter is extra there?